File: 2nd_quarter/week08/Whaeun/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

int arr[7] = {6, 6, 5, 4, 3, 2, 1};

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    map<int, int> win;
    
    
    sort(lottos.begin(), lottos.end());
    
    int zero_cnt = 0;
    int correct_cnt = 0;
    
    sort(win_nums.begin(), win_nums.end());
    
    for(int i = 0; i < win_nums.size(); i++){
        win[win_nums[i]]++;
    }
    
    for(int i = 0; i < lottos.size(); i++){
        if(lottos[i] == 0){
            zero_cnt++;
        }
        else if(win[lottos[i]] > 0){
            correct_cnt++;
        }
    }
    answer.push_back(arr[zero_cnt + correct_cnt]);
    answer.push_back(arr[correct_cnt]);
    
    return answer;
}

File: 2nd_quarter/week08/Royce/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    
    # 당첨 가능한 최고 순위, 최저 순위
    
    num = 0
    cnt = 0
    for k in lottos:
        if k in win_nums:
            num += 1
    # num : 0 제외하고 겹치는 수들
    for k in lottos:
        if k == 0:
            cnt += 1
    if num + cnt == 6:
        answer.append(1)
    elif num + cnt == 5:
        answer.append(2)
    elif num + cnt == 4:
        answer.append(3)
    elif num + cnt == 3:
        answer.append(4)
    elif num + cnt == 2:
        answer.append(5)
    else:
        answer.append(6)
    
    if num == 6:
        answer.append(1)
    elif num == 5:
        answer.append(2)
    elif num == 4:
        answer.append(3)
    elif num == 3:
        answer.append(4)
    elif num == 2:
        answer.append(5)
    else:
        answer.append(6)
    
    return answer
